src/cow-react/modules/swap/services/ethFlow/index.ts

```diff
@@ -0,0 +1,53 @@
+import { EthFlowContext } from '@cow/modules/swap/services/swapFlow/types'
+import { swapFlowAnalytics } from '@cow/modules/swap/services/common/steps/analytics'
+import { signEthFlowOrderStep } from '@cow/modules/swap/services/ethFlow/steps/signEthFlowOrderStep'
+import confirmPriceImpactWithoutFee from 'components/swap/confirmPriceImpactWithoutFee'
+import { logSwapFlow } from '@cow/modules/swap/services/utils/logger'
+import { getSwapErrorMessage } from '@cow/modules/swap/services/common/steps/swapErrorHelper'
+import { PriceImpact } from 'hooks/usePriceImpact'
+import { addPendingEthFLowOrderStep } from '@cow/modules/swap/services/ethFlow/steps/addPendingEthFLowOrderStep'
+
+export async function ethFlow(input: EthFlowContext, priceImpactParams: PriceImpact) {
+  logSwapFlow('ETH FLOW', 'STEP 1: confirm price impact')
+  if (priceImpactParams?.priceImpact && !confirmPriceImpactWithoutFee(priceImpactParams.priceImpact)) return
+
+  logSwapFlow('ETH FLOW', 'STEP 2: send transaction')
+  // TODO: check
+  swapFlowAnalytics.swap(input.swapFlowAnalyticsContext)
+  input.swapConfirmManager.sendTransaction(input.context.trade)
+
+  try {
+    logSwapFlow('ETH FLOW', 'STEP 3: sign order')
+    const { order, orderId } = await signEthFlowOrderStep(input.orderParams, input.contract).finally(() => {
+      input.callbacks.closeModals()
+    })
+
+    logSwapFlow('ETH FLOW', 'STEP 4: add pending order step')
+    addPendingEthFLowOrderStep(
+      {
+        id: orderId,
+        chainId: input.context.chainId,
+        order: {
+          ...order,
+          // TODO: review
+          presignGnosisSafeTxHash: undefined,
```

Comment:
brain hurting not sure if safe presign is applicable for eth-flow orders @josojo

Reply:
Gnosis safes don't need any presign. They can just use the normal onchain order placement for the ethflow. They only needed pre_sign for normal orders, as they can not provide any normal signatures